docs/source/data/fetching-rest.mdx

```diff
@@ -0,0 +1,421 @@
+
+## Creating subclasses
+
+To get started, install the [`@apollo/datasource-rest` package](https://www.npmjs.com/package/@apollo/datasource-rest):
+
+```bash
+npm install @apollo/datasource-rest
+```
+
+Your server should define a separate subclass of `RESTDataSource` for each REST API it communicates with. Here's an example of a `RESTDataSource` subclass that defines two data-fetching methods, `getMovie` and `getMostViewedMovies`:
+<MultiCodeBlock>
+
+```ts title="movies-api.ts"
+import { RESTDataSource } from '@apollo/datasource-rest';
+
+class MoviesAPI extends RESTDataSource {
+  override baseURL = 'https://movies-api.example.com/';
+
+  async getMovie(id):Promise<Movie> {
+    return this.get<Movie>(`movies/${encodeURIComponent(id)}`);
+  }
+
+  async getMostViewedMovies(limit = '10'):Promise<Movie[]> {
+    const data = await this.get('movies', {
+      params: {
+        per_page: limit,
+        order_by: 'most_viewed',
+      },
+    });
+    return data.results;
+  }
+}
+```
+
+</MultiCodeBlock>
+
+You can extend the `RESTDataSource` class to implement whatever data-fetching methods your resolvers need. These methods should use the built-in convenience methods (e.g., `get` and `post`) to perform HTTP requests, helping you add query parameters, parse and cache JSON results, dedupe requests, and handle errors.
+
+## Adding data sources to Apollo Server's context
+
+You can add data sources to the `context` initialization function, like so:
+
+<MultiCodeBlock>
+
+```ts title="index.ts"
+//highlight-start
+interface ContextValue {
+  dataSources: {
+    moviesAPI: MoviesAPI;
+    personalizationAPI: PersonalizationAPI;
+  };
+}
+//highlight-end
+
+const server = new ApolloServer<ContextValue>({
+  typeDefs,
+  resolvers,
+});
+
+const { url } = await startStandaloneServer(server, {
+  context: async () => {
+    const { cache } = server; // highlight-line
+    return {
+      // We create new instances of our data sources with each request,
+      // passing in our server's cache.
+      //highlight-start
+      dataSources: {
+        moviesAPI: new MoviesAPI({ cache }),
+        personalizationAPI: new PersonalizationAPI({ cache }),
+      },
+      //highlight-end
+    };
+  },
+});
+
+console.log(`🚀  Server ready at ${url}`);
+```
+
+</MultiCodeBlock>
+
+<!-- TODO(AS4) add link to context article once exists  -->
+Apollo Server calls [the `context` initialization](../resolvers/#the-context-argument) function for _every incoming operation_. This means:
+- For every operation, `context` returns an _object_ containing new instances of your `RESTDataSource` subclasses (in this case, `MoviesAPI` and `PersonalizationAPI`).
+- The **`context` function should create a new instance of each `RESTDataSource` subclass for each operation.** If multiple operations share a single data source instance, you might accidentally combine results from multiple operations.
```

Comment:
The reason this matters is that in addition to the shared cache described in the next section, there's also a SECOND per-RESTDataSource-object cache that caches *everything* regardless of HTTP headers in-memory.  Which was completely not documented until Shane's PR!  Merging his stuff in will add a bit more of that background but it would be nice if we actually described the two-level cache in the docs explicitly somewhere... see https://github.com/apollographql/apollo-server/pull/6834#pullrequestreview-1079413629

Reply:
Does it feel okay to push back documenting this to the bigger article refresh, so we can merge this faster? If not, I can get on this!

Reply:
I mean it's up to you. I feel like this sentence raises more questions than it answers though... Maybe don't even attempt to explain why you need a new instance for each operation if you're not going to provide the background for anybody to understand?

I am not sure if this is a serious suggestion or not:

> - The **`context` function should create a new instance of each `RESTDataSource` subclass for each operation.** The reason for this is related to the caching data model used by `RESTDataSource`, which is not currently documented but we intend to in the future.

Reply:
Going down that road, what if we just use the first sentence for now? Omit the explanation for the moment until I can expand on this further in a (hopefully fast) follow-up:

> - The **`context` function should create a new instance of each `RESTDataSource` subclass for each operation.**

Reply:
Yeah I think so.